pkg/authorization/cache/readonlycache.go

```diff
@@ -0,0 +1,136 @@
+package cache
+
+import (
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/authorization/client"
+	clusterpolicyregistry "github.com/openshift/origin/pkg/authorization/registry/clusterpolicy"
+	clusterbindingregistry "github.com/openshift/origin/pkg/authorization/registry/clusterpolicybinding"
+	policyregistry "github.com/openshift/origin/pkg/authorization/registry/policy"
+	bindingregistry "github.com/openshift/origin/pkg/authorization/registry/policybinding"
+)
+
+func NewReadOnlyPolicyClient(bindingRegistry bindingregistry.WatchingRegistry, policyRegistry policyregistry.WatchingRegistry,
+	clusterBindingRegistry clusterbindingregistry.WatchingRegistry, clusterPolicyRegistry clusterpolicyregistry.WatchingRegistry) client.ReadOnlyPolicyClient {
+	return NewReadOnlyCache(bindingRegistry, policyRegistry, clusterBindingRegistry, clusterPolicyRegistry)
+}
+
+// ReadOnlyCache embeds four parallel caches for policies and bindings on both the project and cluster level
+type ReadOnlyCache struct {
+	ReadOnlyPolicyCache               readOnlyPolicyCache
```

Comment:
is there a reason you made this public member fields in the struct?  seems like they should all be private fields since interaction with this struct is always through the interface.